specification/msi/resource-manager/Microsoft.ManagedIdentity/2015-08-31-preview/ManagedIdentity.json

```diff
@@ -0,0 +1,544 @@
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Deleted Identity.",
+            "schema": {
+              "$ref": "#/definitions/Identity"
+            }
+          },
+          "204": {
+            "description": "The specified identity does not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Identity": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The id of the created identity."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the created identity."
+        },
+        "location": {
+          "type": "string",
+          "description": "The Azure region where the identity lives."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "The properties associated with the identity.",
+          "$ref": "#/definitions/IdentityProperties"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Microsoft.ManagedIdentity/userAssignedIdentities"
+          ],
+          "x-ms-enum": {
+            "name": "UserAssignedIdentities",
+            "modelAsString": true
+          },
+          "description": "The type of resource i.e. Microsoft.ManagedIdentity/userAssignedIdentities"
+        }
+      },
+      "x-ms-azure-resource": true,
+      "description": "Describes an identity resource."
+    },
+    "IdentityProperties": {
+      "description": "The properties associated with the identity.",
+      "type": "object",
+      "title": "Identity properties.",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "readOnly": true,
+          "format": "UUID",
+          "description": "The id of the tenant which the identity belongs to."
+        },
+        "principalId": {
+          "type": "string",
+          "readOnly": true,
+          "format": "UUID",
+          "description": "The id of the service principal object associated with the created identity."
+        },
+        "clientId": {
+          "type": "string",
+          "readOnly": true,
+          "format": "UUID",
+          "description": "The id of the app associated with the identity. This is a random generated UUID by MSI."
+      },
+        "clientSecretUrl": {
```

Comment:
This should not be a part of response of GET/PUT. Since this is a secret, it should be exposed through a POST action.

Reply:
This is not really secret (bad naming I guess). This is the URL to our data plane from where the user can get credentials.

Reply:
So the question is do you want to expose this properties to anyone who can read the resource or do you want to protect this so only folks with the required permission can read the property?

Reply:
Yes, this should be exposed to anyone who can read the resource.